### PR TITLE
Fix/spatial split - fixes #1

### LINKED ISSUE #1: Implement Spatial Splitting for Model Evaluation
## Description

Implement spatial splitting methodology where entire SNOTEL stations are reserved for testing. This will evaluate model performance on completely unseen locations rather than just unseen time periods.

## Background

Currently using random stratified splitting (70/10/20), where all stations can appear in training/tuning/testing across because I didn't model the autocorrelation. Need to add spatial splitting where ~20% of stations (173 stations) are entirely held out for testing.

### Data Splitting
- [x] Create a spatial split function that reserves entire stations for the test set.
- [x] Ensure stratified sampling by snow climate class for spatial splits, as before.
- [x] Implement 70/10/20 split at station level (not observation level)
- [x] Document split methodology in the methods section

### Model Training & Evaluation
- [x] Retrain all models (RF, Extra Trees, LightGBM, XGBoost) with spatial splits
- [x] Compare performance: temporal vs spatial splitting
- [x] Test all encoding methods (target, CatBoost, one-hot) with spatial splits
- [x] Update hyperparameter tuning for spatial split data

### Results & Analysis
- [x] Update Tables 1-3 with spatial split results

### Documentation
- [x] Update Methods section (Lines 366-378) with spatial split details
- [x] Add discussion of temporal vs spatial prediction objectives

### PR DESCRIPTION
**Pull request by Ibrahim-Ola**
## Description
Implements spatial splitting methodology for hyperparameter tuning where entire SNOTEL stations are reserved for testing, evaluating model performance on completely unseen locations.

## Changes Made
- Updated hyperparameter tuning script to use spatial split data
- Fixed bash script array handling and logging configuration
- Added test modes for validation before full tuning runs
- Integrated spatial split methodology into optimization workflow

## Testing
- [x] Validated spatial split data integration
- [x] Tested hyperparameter tuning with new data splits
- [x] Confirmed logging and bash script functionality

Fixes #1

### CODE REVIEW

File: experiments/hyperparameters.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      low: 6
+      high: 15
+    min_child_weight:
+      type: "int"
+      low: 4
+      high: 50
+    reg_alpha:
+      type: "float"
+      low: 1e-8
+      high: 5.0
```
Comment (Ibrahim-Ola) [inline experiments/hyperparameters.yaml:102]: All looks good! Approving

File: experiments/optimize_ml_models.sh [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+TEST_TRIALS=10
+
+# ===================
+# Helper Functions
+# ===================
+
+run_experiment() {
+    local models="$1"
+    local n_trials=$2
+    local variants="$3"
```
Comment (Ibrahim-Ola) [inline experiments/optimize_ml_models.sh:38]: I like the bash idea!
